### PR TITLE
[BIT] 转换能力补齐：nanquantile、nn.functional.[fold、layer_norm]、tril_indices、triu_indices

### PR DESCRIPTION
**Pull request by yuwu46**
## 修复情况
经回测后，新认领并修复了`nanquantile`、`nn.functional.fold`、`nn.functional.layer_norm`、`tril_indices`、`triu_indices`共5个API，具体情况如下：
| PaddlePaddle API 名称 | PyTorch API 名称 | 功能差异描述 | 配置撰写情况 | 备注 | API 对照表链接 |
|:----------------------|:-----------------|:-------------|:-------------|:-----|:---------------|
| paddle.nanquantile | torch.nanquantile | 参数类型不一致 | 撰写 mapping 配置，复用 rule：`QuantileRule` | Paddle 中的 `axis` 如果是 list，可以对给定的轴上的所有元素计算分位数 | [对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/torch/torch.nanquantile.html) |
| paddle.nn.functional.fold | torch.nn.functional.fold | 参数输入情况不一致 | 撰写 mapping 配置 | torch 中的 `padding` 参数不能为长度是 4 的 list，在 mapping 中对 `padding` 参数做额外转化 | [对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/functional/torch.nn.functional.fold.html) |
| paddle.nn.functional.layer_norm | torch.nn.functional.layer_norm | 参数类型不一致 | 撰写 mapping 配置，同时编写 rule：`LayerNormRule` | 1. 将 `normalized_shape` 转换为 tuple。<br>2. 将 `weight` 和 `bias` reshape 为 `normalized_shape` 的形状。<br>3. 确保 `weight` 和 `bias` 与 `input` 有相同的数据类型。 | [对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/functional/torch.nn.functional.layer_norm.html) |
| paddle.tril_indices | torch.tril_indices | 参数输入情况不一致 | 撰写 mapping 配置 | torch 中的 `col` 不能为 None，在 mapping 中对 `col` 做转化，为 None 时值等于 `row`，与 Paddle 对齐 | [对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/torch/torch.tril_indices.html) |
| paddle.triu_indices | torch.triu_indices | 参数输入情况不一致 | 撰写 mapping 配置 | torch 中的 `col` 不能为 None，在 mapping 中对 `col` 做转化，为 None 时值等于 `row`，与 Paddle 对齐 | [对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/torch/torch.triu_indices.html) |
## 测试结果
本次共从pending文件中移出29条配置，均通过测试，移入`accuracy_8.txt`中。
截屏2025-05-27 17 15 07
## 合并测试结果
将原来已通过的配置和pending配置合并后，共102条，均能通过。
截屏2025-05-27 17 19 46

### CODE REVIEW

File: tester/paddle_to_torch/mapping.json [hunk 1]
```diff
@@ -1247,6 +1263,24 @@
...
+        },
+        "paddle_torch_args_map": {
+            "x": "input",
+            "output_sizes": "output_size",
+            "kernel_sizes": "kernel_size",
+            "strides": "stride",
+            "dilations": "dilation"
+        },
+        "torch_kwargs": {
+            "padding": "({paddings}[0], {paddings}[2]) if isinstance({paddings}, list) and len({paddings}) == 4 else {paddings}"
```
Comment (cangtianhuang) [inline tester/paddle_to_torch/mapping.json:1281]: 实际上变量不使用{}环绕也可以😊～

**Review by cangtianhuang — APPROVED** [review]
LGTM

**Review by wanghuancoder — APPROVED** [review]
LGTM
